### ISSUE TITLE
unstable_shouldReload not working with deferred in some cases

### ISSUE THREAD

**Issue by rhefner** [remix-run/remix#3712]
### What version of Remix are you using?

0.0.0-experimental-d28d77b8

### Steps to Reproduce

[Codesandbox showing the issue](https://codesandbox.io/s/remix-deferred-unstable-shouldreload-issue-tf12d5)

Steps to see the potential issue:
- Go to "root" route
- Click on "Deferred" link
- Click the button in the `<nav />` to toggle and save the cookie
- Notice two fetches to /deferred:
```
/deferred?_data=root
/deferred?_data=routes%2Fdeferred
```

Steps to see `unstable_shouldReload` working (I think):
- Go to "/deferred" route directly
- Click on the cookie setter button
- Notice one fetch to /deferred:
```
/deferred?_data=root
```

### Expected Behavior

`unstable_shouldReload` should be respected in all cases - Whether or not it's the first route the user visits or subsequent ones

### Actual Behavior

`unstable_shouldReload` is not respected if the user doesn't go directly to the route

**Comment by MichaelDeBoey** [comment]
Closed by #3721

**Comment by brophdawg11** [comment]
Really digging the symmetry of `3721` closing `3712` - I thought it was a typo at first 😂
